plugins/aws_ecr/lib/samson_aws_ecr/samson_plugin.rb

```diff
@@ -3,29 +3,57 @@
 module SamsonAwsEcr
   class Engine < Rails::Engine
     AMAZON_REGISTRY = /\A.*\.dkr.ecr.(?<region>[\w\-]+).amazonaws.com\z/
-    mattr_accessor :ecr_client
-    mattr_accessor(:credentials_expire_at) { Time.current }
 
-    initializer :ecr_client do
-      if match = AMAZON_REGISTRY.match(ENV['DOCKER_REGISTRY'])
-        SamsonAwsEcr::Engine.ecr_client = Aws::ECR::Client.new(region: match['region'])
+    class << self
+      # we make sure the repo exists so pushes do not fail
+      # - ignores if the repo already exists
+      # - ignores if the repo cannot be created due to permission problems
+      def ensure_repository(repository)
+        if ecr_client
+          ecr_client.create_repository(
+            repository_name: repository.split('/', 2).last
+          )
+        end
```

Comment:
Could you actually do a check here if the registry already exists?

We do monitoring of our CloudTrail logs, and I think we look for failed API calls. Would be nice if we didn't have a whole bunch of false positive failures because Samson keeps trying to create a repo that already exists.